source/extensions/filters/http/istio_stats/istio_stats.cc

```diff
@@ -457,6 +458,73 @@ struct MetricOverrides : public Logger::Loggable<Logger::Id::filter>,
   absl::flat_hash_map<std::string, uint32_t> expression_ids_;
 };
 
+// Self-managed scope with active rotation. Envoy stats scope controls the
+// lifetime of the individual metrics. Because the scope is attached to xDS
+// resources, metrics with data derived from the requests can accumulate and
+// grow indefinitely for long-living xDS resources. To limit this growth,
+// this class implements a rotation mechanism, whereas a new scope is created
+// periodically to replace the current scope.
```

Comment:
- What is the reason behind "a new scope is created periodically to replace the current scope"? As scope controls the lifetime of a metric, it seems an original scope (not a newly created scope) should be associated with a metric.
- If "a new scope is created periodically to replace the current scope", when will a metric expire and be cleared to save memory?

Reply:
A metric (a set of symbols for the name and the tags) is cleared when its parent reference count reaches 0.
A new scope is needed to emit new metrics - this is happening in active listener handling requests.
An old scope is kept for a short window (both old and new are present then), as explained in the comment, before deleting it and all its metrics (that are not used elsewhere).